game/src/game/components/state/PlayerState.cpp

```diff
@@ -0,0 +1,37 @@
+#include "PlayerState.h"
+
+#include <iostream>
+
+void PlayerState::OnInit(const ServiceProvider& service_provider)
+{
+    GetEventBus().Subscribe<OnUpdateEvent>(this);
+    game_state_service_ = &service_provider.GetService<GameStateService>();
+}
+
+void PlayerState::OnStart()
+{
+}
+
+void PlayerState::OnUpdate(const Timestep& delta_time)
+{
+    if (!game_state_assigned_)
+    {
+        if (attached_entity_)
+        {
+            game_state_service_->AddPlayerDetails(attached_entity_,
+                                                  player_state_);
+            // so that next time we do not enter this loop and redundantly add
+            // more in the player details
+            game_state_assigned_ = true;
+        }
+    }
```

Comment:
Why not do this in OnInit instead? Especially if this is paired with the other change I suggested of not having a settable entity, we can just do this immediately and forget about it